### PR TITLE
feat(specs): add shared JSON Schema validation for honeypot configurations

### PR DESCRIPTION
**Pull request by gzigurella**
All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### New Feature Submissions:

1. [x] Does your submission pass tests?
2. [x] Have you lint your code locally before submission?

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

---

## Summary

Adds a **shared JSON Schema validation layer** for Beelzebub honeypot service configurations. The schemas in `specs/` define the canonical validation contract consumed by the Go runtime, and designed to be consumed by other platforms (Java backend, TypeScript dashboard).

### New files

| File | Description |
|---|---|
| `specs/honeypot-config.schema.json` | Base schema (draft-07) with `$defs` for Command, Tool, Plugin, tool annotations, and conditional `if/then` rules for LLM, Maze, and rate limiting |
| `specs/honeypot-ssh.schema.json` | SSH: requires `passwordRegex`, `serverVersion`, `commands` |
| `specs/honeypot-http.schema.json` | HTTP: requires `commands`, disallows `tools` |
| `specs/honeypot-tcp.schema.json` | TCP: disallows `tools` |
| `specs/honeypot-telnet.schema.json` | TELNET: requires `passwordRegex`, `commands` |
| `specs/honeypot-mcp.schema.json` | MCP: requires `tools`, disallows `commands` |
| `specs/embed.go` | Embeds schemas into the Go binary via `//go:embed` |
| `internal/parser/schema_validator.go` | `SchemaValidator` — registered as `ServiceValidator`, lazy-compiles schemas with `sync.Once`, validates via `santhosh-tekuri/jsonschema/v6` |
| `internal/parser/schema_validator_test.go` | 20 test cases: valid configs per protocol, invalid edge cases, unknown protocol, error flattening |
| `cmd/validate-specs/main.go` | Standalone CLI for CI (`make validate-specs`) |

### Modified files

| File | Change |
|---|---|
| `internal/parser/configurations_parser.go` | Added JSON tags with `omitempty` to all struct fields (Plugin, Command, Tool, etc.) for correct schema serialization |
| `internal/parser/validator.go` | Removed `validateProtocol()` (schema enum handles it), removed `validCommandPlugins` duplicate checks |
| `cli/validate_test.go` | Updated expected error count for unknown protocol (schema + Go → schema only) |
| `internal/parser/validator_test.go` | Updated tests to register `SchemaValidator` via `RegisterServiceValidator()` for proper pipeline testing |
| `internal/parser/configurations_parser_test.go` | Updated hash codes for new omitempty JSON output, fixed JSON keys (`RegexStr` → `regex`), added `hasIssueContaining` helper |
| `internal/plugins/beelzebub-cloud_test.go` | Updated hash code for new JSON output |
| `HTTP/validator_test.go` | Cross-platform TLS tests via `os.Executable()` instead of `/proc/self/exe` |
| `TCP/validator_test.go` | Cross-platform TLS tests via `os.Executable()` |
| `Makefile` | Added `validate-specs` and `validate-all` targets |
| `CONTRIBUTING.md` | Added "Configuration Validation Schema" section with Mermaid sequence diagram and 38-row rule reference table |
| `go.mod` | `santhosh-tekuri/jsonschema/v6` promoted from indirect to direct dependency |

### Validation architecture

YAML → ReadConfigurationsServicesForValidation()
        → yaml.Unmarshal + regex.Compile + net.ParseCIDR
        → Validate() loop:
            ├── Direct Go checks (address, commands, fallback, plugin config)
            ├── SchemaValidator (lazy-compiled JSON Schema via jsonschema/v6)
            │   └── json.Marshal → validate per-protocol schema
            └── Protocol-specific validators (SSH, HTTP, TCP, TELNET, MCP)
        → detectCollisions()
        → ValidateResult{errors + warnings}

### Dual-layer design

- **Schema** (JSON Schema draft-07) — structural errors: required fields, types, enums, patterns, `if/then` conditionals for LLM/Maze/rate limiting
- **Go procedural** — quality warnings (inline secrets, missing deadline/timeout, empty handlers) and runtime checks (regex compile, CIDR parse, TLS file existence, collision detection)

### CODE REVIEW

**Comment by airscripts** [comment]
Hey @gzigurella I did a deep review of this PR and I only found one potentially blocking finding.  
I've also listed a few lower prio since it's the open runtime, so checks have to be strictier.  
As always we can discuss them if they are out of scope.

```md
## Findings

### 1. High — Schema validation does not validate the original YAML

This branch introduces the new validation flow where cmd/validate-specs/main.go:73 unmarshals YAML into BeelzebubServiceConfiguration, and internal/parser/schema_validator.go:127
serializes that struct back to JSON for schema validation.

This loses information that the schemas added by this branch are intended to validate:

- Unknown or misspelled properties disappear, defeating additionalProperties: false.
- Explicit zero values disappear through the new omitempty JSON tags.
- Omitted fields cannot be distinguished from explicitly supplied zero values.
- Constraints such as statusCode being between 100 and 599 can be bypassed.

I confirmed that the new validate-specs command accepts a configuration containing an unknown commmands property.

Recommendation: validate a JSON-compatible representation of the original YAML document before unmarshalling it into the runtime Go structure. Add end-to-end coverage for unknown
fields, nested typos, explicit zero values, empty collections, and omitted-versus-empty fields.

### 2. Medium — The branch accidentally downgrades the RabbitMQ dependency

Relative to the actual merged-main parent, this branch changes go.mod:13 from:

github.com/rabbitmq/amqp091-go v1.13.0

to:

github.com/rabbitmq/amqp091-go v1.12.0

The matching go.sum entries are also downgraded. This is unrelated to ENG-221 and appears to be an accidental merge resolution.

Recommendation: restore v1.13.0 and its checksums.

### 3. Medium — Move the validation documentation added to CONTRIBUTING.md

This branch adds approximately 150 lines of validation architecture to CONTRIBUTING.md:103, including the validation-flow diagram, schema architecture, protocol rules, commands,
and validation matrix.

Most of this is product and architecture documentation, not contribution guidance.

Recommendation: move it into a dedicated document such as:

docs/configuration-validation.md

Keep only a short contributor-oriented summary in CONTRIBUTING.md, explaining what contributors must update when changing configuration fields or validators.

This branch currently does not change README.md relative to the merged main parent. It should add a link there to the new document, preferably near beelzebub validate or the
configuration reference.

### 4. Low — apiVersion documentation, schema, and tests disagree

The rule table added by this branch says apiVersion must be "v1" at CONTRIBUTING.md:255.

The new schema constrains the value at specs/runtime-config.schema.json:8, but its required list at specs/runtime-config.schema.json:57 contains only:

"required": ["protocol", "address"]

Tests added by this branch also treat configurations without apiVersion as valid.

Recommendation: decide whether apiVersion is mandatory:

- If mandatory, add it to required and add a missing-version failure test.
- If optional for compatibility, update the documentation accordingly.

### 5. Low — The new command documents a flag that does not exist

The package documentation in cmd/validate-specs/main.go:9 advertises:

go run ./cmd/validate-specs -specs path/to/specs

However, the command only defines -configs; there is no -specs flag. Running the documented command returns an unknown-flag error.

Recommendation: either implement -specs or remove that usage example. Since schemas are embedded, removing the example is likely the correct choice.
```

**Comment by gzigurella** [comment]
@airscripts 

My bad, `-specs` flag remaind committed on my local, `-configs` defines the honeypot configuration is checking, while `-specs` what rules are used for the check.

The same goes for apiVersion as well, both worked on the 29 of July.

Taking care of reviews 1,2 and 3, then I'll push them as well 🚀 omw!

File: specs/runtime-config.schema.json [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+      "pattern": "^(\\[[^\\]]+\\]|[^:\\s]+)?:[0-9]{1,5}$|^/[^:]+$",
+      "description": "Listen address in host:port format (e.g. :22, 0.0.0.0:8080, [::1]:443). Unix sockets use /path format."
+    },
+    "description": { "type": "string", "description": "Human-readable label for this honeypot service." },
+    "serverVersion": { "type": "string", "description": "Banner/version string sent to clients during handshake (e.g. OpenSSH, Apache/2.4.53)." },
+    "serverName": { "type": "string", "description": "Hostname reported by the honeypot (e.g. ubuntu, mail.internal.company.com)." },
+    "deadlineTimeoutSeconds": { "type": "integer", "minimum": 0, "description": "Connection deadline in seconds for TCP, TELNET, and SSH. HTTP does not use this field; 0 causes immediate deadlines for TCP and TELNET, while SSH treats 0 as no timeout." },
+    "banner": { "type": "string", "description": "Initial banner sent on connection (TCP) or greeting (HTTP). Can include binary wire format." },
+    "passwordRegex": { "type": "string", "description": "Regex pattern for accepted passwords. Required for SSH and TELNET. Syntax validated by Go at runtime." },
+    "trustedProxies": {
```
Comment (mariocandela) [inline specs/runtime-config.schema.json:29]: Missing 'trustedProxiesNets' :)

**Review by mariocandela — DISMISSED** [review]

File: specs/runtime-config.schema.json (hunk 1 above)
Comment (gzigurella) [reply specs/runtime-config.schema.json:29]: The user-facing configuration field is trustedProxies, according to the history of the repo (499c181 treated TrustedProxiesNets as the internal state, and TrustedProxices as the user-facing, right?).

The parser compiles it into TrustedProxiesNets before runtime. This PR validates the user-facing configuration, in order to share it with the cloud platform, not the internal runtime representation. TrustedProxiesNets was already an internal derived field in the existing trusted-proxy implementation, tagged with yaml:"-" json:"-", am I missing something?

Also closed the gaps around the parser and added some test for better coverage of edge cases in my new commit [1b0c926](https://github.com/beelzebub-labs/beelzebub/pull/341/commits/1b0c9267cb1b79da105b6515fcf28eb8733f862e)

**Review by mariocandela — APPROVED** [review]
https://lgtm.airscript.it/lgtm/11

**Comment by mariocandela** [comment]
Awesome work @gzigurella <3

**Review by airscripts — APPROVED** [review]
Nicely done!
P.S.: https://lgtm.airscript.it/lgtm/2
